connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,42 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ConnectionTest {
```

Comment:
1. If you can't mock its dependencies then you don't fully understand how they function / interact together.
The reason you have troubles mocking an InputStream is ObjectInputStream. Upon a construction it tries to read some header (or block the thread it runs in), verify it and then it is ready to do other things. You can jump in the code (decompiled by IDEA) to look at it closer.
I would suggest to not use ObjectInputStream as it blocks at a construction time and its not a very testable / nice feature to have. Find a better message serializer.
If you still want to use it you will have to test it like [this](https://github.com/shouldDoFine/Kirill/tree/feature/example_of_involved_test) (I, out of curiosity, created one possible test).
2. Still, its not an excuse for writing nonsense :)